inc/datahandlers/user.php

```diff
@@ -868,6 +868,31 @@ function verify_language()
 	}
 
 	/**
+	 * Verifies if a style is valid for this user or not.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_style()
+	{
+		global $lang;
+
+		$user = &$this->data;
+
+		if($user['style'] && !defined('IN_ADMINCP'))
+		{
+			$theme = get_theme($user['style']);
+
+			if(empty($theme) || !is_member($theme['allowedgroups'], array('usergroup' => $user['usergroup'], 'additionalgroups' => $user['additionalgroups'])) && $theme['allowedgroups'] != 'all')
```

Comment:
As I said here: https://github.com/mybb/mybb/pull/2052 I don't mind either way, but I'd also prefer this.